### PR TITLE
[New Rule] AWS Redshift Cluster Creation

### PR DESCRIPTION
**Pull request by pjhampton**
## Issues

Closes: https://github.com/elastic/detection-rules/issues/1918

## Summary
Adds a rule around data warehouse creation within AWS. 

```
event.dataset:aws.cloudtrail 
    and event.provider:redshift.amazonaws.com 
    and event.action:CreateCluster 
    and event.outcome:success
```

Screenshot 2022-04-12 at 12 53 03

https://gist.github.com/pjhampton/61a8ba69a1d98e3c55e15f2f6cf9bf4f

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: rules/integrations/aws/persistence_redshift_instance_creation.toml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+[metadata]
+creation_date = "2022/04/12"
+maturity = "production"
+updated_date = "2022/04/12"
+integration = "aws"
+
+[rule]
+author = ["Elastic", "Pete Hampton"]
+description = "Identifies the creation of an Amazon Redshift cluster."
```
Comment (w0rk3r) [inline rules/integrations/aws/persistence_redshift_instance_creation.toml:9]: We could elaborate a bit on why it is important to monitor this, usages on the attacker's perspective, etc.

File: rules/integrations/aws/persistence_redshift_instance_creation.toml [hunk 2]
```diff
@@ -0,0 +1,45 @@
+[metadata]
+creation_date = "2022/04/12"
+maturity = "production"
+updated_date = "2022/04/12"
+integration = "aws"
+
+[rule]
+author = ["Elastic", "Pete Hampton"]
```
Comment (w0rk3r) [inline rules/integrations/aws/persistence_redshift_instance_creation.toml:8]:
```suggestion
author = ["Elastic"]
```
We usually only include authors other than Elastic in community contributions

File: rules/integrations/aws/persistence_redshift_instance_creation.toml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    """,
+]
+from = "now-60m"
+index = ["filebeat-*", "logs-aws*"]
+interval = "10m"
+language = "kuery"
+license = "Elastic License v2"
+name = "AWS Redshift Cluster Creation"
+note = """## Config
+The AWS Fleet integration, Filebeat module, or similarly structured data is required to be compatible with this rule."""
```
Comment (w0rk3r) [inline rules/integrations/aws/persistence_redshift_instance_creation.toml:24]:
```suggestion
note = """## Config

The AWS Fleet integration, Filebeat module, or similarly structured data is required to be compatible with this rule."""
```

**Review by w0rk3r — CHANGES_REQUESTED** [review]
Thanks for the PR, left a few comments/requests

File: rules/integrations/aws/persistence_redshift_instance_creation.toml (hunk 1 above)
Comment (pjhampton) [reply rules/integrations/aws/persistence_redshift_instance_creation.toml:9]: The rule I really want to write is around the networking around redshift (a dw that we are likely to use soon) - https://docs.aws.amazon.com/redshift/latest/APIReference/API_AuthorizeClusterSecurityGroupIngress.html
However, surveying your other rules, particularly around RDS clusters / instances, there is rules around the creation and deletion of the data stores.

Should I close this and just write the security group ingress changes?

**Review by w0rk3r — APPROVED** [review]
LGTM once we elaborate a bit the description

**Comment by terrancedejesus** [comment]
> 

Since this is a little less about a specific intrusion playbook and more about proper management and awareness of cloud resources. What do you think about the following? We can get this simple rule wrapped up and merged if agreed.

```
Identifies the creation of an Amazon Redshift cluster. Unexpected creation of this cluster by a non-administrative user may indicate a permission or role issue with current users. If unexpected, the resource may not properly be configured and could introduce security vulnerabilities.
```

**Comment by pjhampton** [comment]
Thanks, @terrancedejesus and @w0rk3r!

> Since this is a little less about a specific intrusion playbook and more about proper management and awareness of cloud resources.

That right. SDE mainly uses Elastic Security to monitor and encourage AWS best practices across our cloud accounts, and encourage secure coding / infra. Do you folks rather not receive rules like this, and those that focus on known attack behaviours? 

---

Also, thanks both. I have learned a lot about how these are developed and this will help me improve the telemetry for you folks with my team.

**Comment by terrancedejesus** [comment]
> Thanks, @terrancedejesus and @w0rk3r!
> 
> > Since this is a little less about a specific intrusion playbook and more about proper management and awareness of cloud resources.
> 
> That right. SDE mainly uses Elastic Security to monitor and encourage AWS best practices across our cloud accounts, and encourage secure coding / infra. Do you folks rather not receive rules like this, and those that focus on known attack behaviours?
> 
> Also, thanks both. I have learned a lot about how these are developed and this will help me improve the telemetry for you folks with my team.

Thanks for the contribution and help! As you stated best practices is important to monitor for and can sometimes help identify a small but important breadcrumb left behind during an intrusion so using them during triaging efforts can potentially help.

Based on my understanding, our focus on rules is more about coverage than anything. So any new rule for a gap in visibility or a specific TTP is important where the customer/client can then tailor their usage of the SIEM according to their needs which is why they can enable/disable, add exception lists, etc. We also have promotion rules that help ingest alerts from another product, such as MS Sentinel, into Elastic so those signals are not missed as well.

**Pull request merged**